types/index.esm.d.ts

```diff
@@ -2315,26 +2315,34 @@ export interface ExtendedPlugin<
    */
   afterTooltipDraw?(chart: Chart, args: { tooltip: Model }, options: O): void;
 }
+
+export interface ScriptableTooltipContext<TType extends ChartType> {
+  chart: Chart;
+  tooltip: TooltipModel<TType>;
+  tooltipItems: TooltipItem<TType>[];
+}
+export type ScriptableTooltip<T, TType> = T | (TType extends ChartType ? { [TT in TType]: ((ctx: ScriptableTooltipContext<TT>) => T) }[TType] : ((ctx: TType) => T));
```

Comment:
What is the reasoning/need for this? I'd expect to use `Scriptable` with the tooltip context.
Previously when `Scriptable` had the same ternary, it turned to `any` wth when TType was union. Doesn't this do the same?

Reply:
Initially I tried using `Scriptable<number, ScriptableTooltipContext <TType>>` but it broke the assignment tests. Doing this fixed it and I believe the reason is because the input takes `TType` rather than `ScriptableTooltipContext<TType>`

Reply:
If you also use the `UnionToIntersection` helper in the ScriptableTooltipContext, then the assignment test should pass. Not sure if its needed for all fields. I'm unaware of the drawbacks of the helper though :)

Reply:
Will give that a shot tonight

Reply:
Thought of a reason against doing this. When you call `setContext`, the options get deeper because we wrap the existing `Proxy` in a new `Proxy`. This would lead to a memory leak in the tooltip because it updates so often.

Reply:
Ok, tried `UnionToIntersection` locally and this is where is dies.

```
 Type '(ctx: ScriptableTooltipContext<"scatter">) => boolean' is not assignable to type '(ctx: ScriptableTooltipContext<"line" | "bar" | "bubble" | "doughnut" | "polarArea" | "pie" | "radar" | "scatter">) => boolean'.
```

Reply:
For reference, the `ScriptableTooltipContext` is

```ts
export interface ScriptableTooltipContext<TType extends ChartType> {
  chart: UnionToIntersection<Chart<TType>>;
  tooltip: UnionToIntersection<TooltipModel<TType>>;
  tooltipItems: UnionToIntersection<TooltipItem<TType>[]>;
}
```

Reply:
Got it passing by updating `TooltipItem` to use `UnionToIntersection` instead of the context directly.